config/config-api/src/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -358,12 +360,14 @@ public void updateGroup(PipelineConfigs pipelineConfigs, String groupName) {
 
         @Override
         public ConfigOrigin getOrigin() {
-            MergeConfigOrigin origins = new MergeConfigOrigin(this.main.getOrigin());
-            for(PartialConfig part : this.parts)
-            {
-                origins.add(part.getOrigin());
-            }
-            return origins;
+            return new MergeConfigOrigin();
+            //TODO: jyoti, check if we still need this
```

Comment:
Probably we don't need this. IIRC there is no code that actually reads those partial origins from MergeConfigOrigin instances. 
However I added and implemented `MergeConfigOrigin` to help with debugging, after all it does give you some idea of what are the sources of parts in merged instance.
You can remove it if you like, `return new MergeConfigOrigin();` should work fine. There is a failing test because of this `shouldReturnOriginAsASumOfAllOrigins`